### PR TITLE
GHC Head updates

### PR DESCRIPTION
**Pull request by mpickering**
###### Things done

Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers.

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/ghc/head.nix [hunk 1]
```diff
@@ -5,15 +5,15 @@
   # If enabled GHC will be build with the GPL-free but slower integer-simple
   # library instead of the faster but GPLed integer-gmp library.
 , enableIntegerSimple ? false, gmp
+, version ? "8.3.20170808"
```
Comment (cocreature) [inline pkgs/development/compilers/ghc/head.nix]: Allowing to override the version number but not the actual version that’s used, i.e., `src` seems a bit weird.

Comment (mpickering) [inline pkgs/development/compilers/ghc/head.nix]: I could override `src` with `overrideDerivation` but if I remember right, I couldn't do the same for `version`.

Comment (Ericson2314) [inline pkgs/development/compilers/ghc/head.nix]: I wish there was a way for mkDerivation to take a fixed point...

**Comment by mpickering** [comment]
The build seems to work correctly with these changes.

**Pull request merged**